xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-resource/entityResourcePicker.js

```diff
@@ -181,7 +181,15 @@ define('entityResourcePicker', [
       }
     });
     handler.open = function(resourceReference) {
-      entityTreePickerHandler.open($resource.convertResourceReferenceToEntityReference(resourceReference));
+      // If the editor instance has a current instance defined, use it as the base entity reference to resolve the 
+      // resource reference.  
+      var baseEntityReference;
+      var currentInstance = CKEDITOR.currentInstance;
+      if (currentInstance) {
```

Comment:
This is the part that I don't like. The resource picker should not be aware of any editor. It's simply a widget that allows you to pick a resource from the wiki. It's not much different, conceptually, than a date picker. It just happens that the CKEditor has the need to select a resource and it uses the resource picker for this. But the Wiki editor could use it too, or some other module. I know it is packaged with CKEditor, but this was needed when the code was in contrib because we were supporting older versions of XWiki so we couldn't have this resource picker in XWiki. Now we can move it in a separate WebJar module if we want.

I would prefer instead to:
* pass the current document as a parameter to the resource picker here https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-14.10.1/xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-resource/plugin.js#L121 , besides the ``resourceTypes`` parameter. I think "current" or "base" makes more sense for the resource picker than "source" (``sourceDocument`` makes sense for the editor)
* store the resource picker parameters (options) as (jQuery) data on the resource picker element, here https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-14.10.1/xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-resource/resourcePicker.js#L47-L50
* take the parameters / options from the resource picker element and pass the "current" / "base" document reference when opening the dedicated pickers here https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-14.10.1/xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-resource/resourcePicker.js#L184 , besides the ``resourceReference``
* use the "current" / "base" document reference in the dedicated pickers here https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-14.10.1/xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-resource/entityResourcePicker.js#L216 , with a fallback to ``XWiki.currentDocument``
* find all occurrences of ``XWiki.currentDocument`` in dedicated resource pickers and replace with the usage of the passed "current" / "base" document reference.

Reply:
I've pushed a new version that, I hope, follows your recommendations.
The resources are now selected, and resolve, relatively to the current document of the editor if it exists.